lib/schemas/payment.js

```diff
@@ -0,0 +1,93 @@
+const {config} = require('bedrock');
+const {PaymentStatus} = require('bedrock-payment');
+
+const w3cDateTime = require('bedrock-validation/schemas/w3cDateTime');
+const numericFloat = '^[0-9]+(\\.[0-9]+)?$';
+const paymentEnum = Object.values(PaymentStatus);
+// this is used by the validator to ensure you
+// can not post a service not supported by the current project.
+const servicesEnum = config.bedrock_payment.services || [];
+
+const _payment = ({required}) => ({
+  title: 'Bedrock Payment',
+  type: 'object',
+  required,
+  additionalProperties: false,
+  properties: {
+    id: {
+      title: 'id',
+      type: 'string',
+    },
+    amount: {
+      title: 'amount',
+      type: 'string',
+      pattern: numericFloat
+    },
+    currency: {
+      title: 'currency',
+      type: 'string'
+    },
+    creator: {
+      title: 'creator',
+      type: 'string'
+    },
+    validated: {
+      title: 'validated',
+      anyOf: [{type: 'boolean'}, {type: 'null'}]
+    },
+    service: {
+      title: 'service',
+      name: 'service',
+      type: 'string',
+      enum: servicesEnum
+    },
+    serviceId: {
+      title: 'serviceId',
+      type: 'string'
+    },
+    status: {
+      title: 'status',
+      type: 'string',
+      enum: paymentEnum
+    },
+    error: {
+      title: 'error',
+      anyOf: [{type: 'object'}, {type: 'null'}]
+    },
+    orders: {
+      title: 'orders',
+      type: 'array',
+      minItems: 1
```

Comment:
Is there a schema for the array `items` here?

Reply:
no this is intentionally left open so we can support a variety of order types.

Reply:
What are some possible order types? Do we need to handle different order types with different logic paths?

Reply:
Possible order types:

1. Pay for a plan once.
2. Subscriptions for a Plan.
3. Charge when needed i.e. an account's balance is up so they automatically recharge.
4. Whatever veres-wallet will be using this for.

The orders array is passed to the order processor. I believe the major concern with orders is simply just the size of it in the database. More than open to a discussion on what this data structure should be. At the moment for the accelerator it holds a string for the single plan you purchase.

Reply:
It feels like there is a general data model we can deduce from this. We should find time to sort out what that is and ensure we fold that into this library. After, we should update the dependent libraries as well. We can plan time to work on this over voice and schedule something over chat.

Reply:
@gannan08 this branch now contains the updated data model. Validators have been updated to: https://github.com/digitalbazaar/bedrock-payment-http/pull/1/commits/8fca7d975f4cc829af1f6a28222b6ca7b75405d2